### ISSUE TITLE
Typescript `eslint --init` false positives

### ISSUE THREAD

**Issue by cdalexndr** [eslint/eslint#13003]
**Tell us about your environment**

* **ESLint Version:** 6.8.0
* **Node Version:** 10.16.0
* **npm Version:** 6.13.1

**What parser (default, Babel-ESLint, etc.) are you using?**
@typescript-eslint/parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
env:
  browser: true
  es6: true
extends:
  - 'eslint:recommended'
  - 'plugin:@typescript-eslint/eslint-recommended'
globals:
  Atomics: readonly
  SharedArrayBuffer: readonly
parser: '@typescript-eslint/parser'
parserOptions:
  ecmaVersion: 2018
  sourceType: module
plugins:
  - '@typescript-eslint'
rules: {}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**
I run the initialization wizard with the following options:
```
λ npx eslint --init
? How would you like to use ESLint? To check syntax and find problems
? What type of modules does your project use? JavaScript modules (import/export)
? Which framework does your project use? None of these
? Does your project use TypeScript? Yes
? Where does your code run? Browser
? What format do you want your config file to be in? YAML
The config that you've selected requires the following dependencies:

@typescript-eslint/eslint-plugin@latest @typescript-eslint/parser@latest
? Would you like to install them now with npm? Yes
```

**What did you expect to happen?**
To **only** use typescript rules for `.ts` files.

**What actually happened? Please include the actual, raw output from ESLint.**
Typescript lint errors like: https://github.com/typescript-eslint/typescript-eslint/issues/1679

**Are you willing to submit a pull request to fix this bug?**
No

**Comment by cdalexndr** [comment]
Maybe the generated file should also include [plugin:@typescript-eslint/recommended](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/configs/recommended.json)?
This disables the `no-unused-vars` false positive with:
`"no-unused-vars": "off"`

**Comment by kaicataldo** [comment]
Extending `"plugin:@typescript-eslint/recommended"` makes a lot of sense to me :+1:. This config [overrides core rules as necessary](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin#recommended-configs), and would provide a better user onboarding experience.

Regarding the proposal in the issue, I do think this would make sense. It would also be nice to print a message to the user letting them know that they must use the `--ext` option to lint files that don't end in the `.js` extension.

I'm envisioning generating a config with the following general structure for TS users:

```js
{
  extends: "eslint:recommended",
  overrides: [
    {
      files: ["*.ts", "*.tsx"]
      extends: [
        "plugin:@typescript-eslint/eslint-recommended"],
        "plugin:@typescript-eslint/recommended"
    }
}
```
